### DISCUSSION TITLE
Security Onion 2.3.10 installation freezes at "Applying thehive Salt State"

### DISCUSSION THREAD

**Discussion by w0lfcat** [Security-Onion-Solutions/securityonion#2399] [Unsupported Versions] [answered]
I had an issue with SO installation earlier, so I decided to install it again for 5 or 6 times ( don't remember anymore ) but now the installation stuck at "Applying thehive Salt State". It's been like this for a few hours now.

![image](https://user-images.githubusercontent.com/71960097/102710583-2aa46a80-42ee-11eb-9ae7-d0543c010638.png)

I ssh to the box and this is what I found on `/root/sosetup.log`

```
[admin@securityonion ~]$ sudo tail /root/sosetup.log

 so-filebeat
 so-wazuh
 so-thehive-es
+so-cortex
[INFO    ] Completed state [/opt/so/conf/so-status/so-status.conf] at time 07:34:38.532039 (duration_in_ms=12.085)
[INFO    ] Running state [cortexscript] at time 07:34:38.534005
[INFO    ] Executing state cmd.script for [cortexscript]
[INFO    ] Fetching file from saltenv 'base', ** done ** 'thehive/scripts/cortex_init'
[INFO    ] Executing command '/opt/so/__salt.tmp.loscdtdv' in directory '/opt/so'
[admin@securityonion ~]$ 

```
If there is any solution for this, please let me know.

**More info**

Exact version. ex. 2.3.10
Install source: Security Onion ISO image in VirtualBox.
Install type: eval
Does so-status show all services running?

```
[admin@securityonion ~]$ sudo so-status
[sudo] password for admin: 

Checking Docker status

    Docker -------------------------------------------------- [ OK ]    

Checking container statuses

    so-cortex --------------------------------------- [ WAIT_START ]    
    so-curator -------------------------------------- [ WAIT_START ]    
    so-dockerregistry ------------------------------- [ WAIT_START ]    
    so-elasticsearch -------------------------------- [ WAIT_START ]    
    so-filebeat ------------------------------------- [ WAIT_START ]    
    so-fleet ---------------------------------------- [ WAIT_START ]    
    so-idstools ------------------------------------- [ WAIT_START ]    
    so-kibana --------------------------------------- [ WAIT_START ]    
    so-kratos --------------------------------------- [ WAIT_START ]    
    so-mysql ---------------------------------------- [ WAIT_START ]    
    so-nginx ---------------------------------------- [ WAIT_START ]    
    so-playbook ------------------------------------- [ WAIT_START ]    
    so-redis ---------------------------------------- [ WAIT_START ]    
    so-sensoroni ------------------------------------ [ WAIT_START ]    
    so-soc ------------------------------------------ [ WAIT_START ]    
    so-steno ---------------------------------------- [ WAIT_START ]    
    so-suricata ------------------------------------- [ WAIT_START ]    
    so-thehive-es ----------------------------------- [ WAIT_START ]    
    so-wazuh ---------------------------------------- [ WAIT_START ]    
    so-zeek ----------------------------------------- [ WAIT_START ]    

[admin@securityonion ~]$ 
```

Do you get any failures when you run salt-call state.highstate from the node? Yes

```
[admin@securityonion ~]$ salt-call state.highstate
[ERROR   ] Could not cache minion ID: [Errno 13] Permission denied: '/etc/salt/minion_id'
[WARNING ] Failed to open log file, do you have permission to write to /var/log/salt/minion?
[ERROR   ] DNS lookup or connection check of 'salt' failed.
[ERROR   ] Master hostname: 'salt' not found or not responsive. Retrying in 30 seconds
```

**Comment by Sazerak** [comment]
I just recently had this issue and in my case, it was due to a misconfigured network. Basically, the default route that I had setup initially through ESXi was using the monitoring interface, rather than the management interface. If you do sudo cat /root/sosetup.log | grep "ERROR" do you get lots of errors? Possibly with many related to MySQL? That's what happened to me.

Maybe start by checking your network configuration and ensure that the interface you want to be your monitoring interface isn't your default route before you start the installation. If it is, either flip your choices in the config wizard or change the configurations on your cards and restart the configuration. Once I did that, everything went well.

My old post is [HERE](https://github.com/Security-Onion-Solutions/securityonion/discussions/2382) if you want to check it out.

**Reply by w0lfcat** [reply]
Thanks, I don't have MySQL error in this case.

**Comment by nicolas-gagnon** [comment]
I facing the same issue, have you resolved your  problem ?

**Reply by w0lfcat** [reply]
Too bad, I've no idea what is going on with this. I've tried so many times, with different configuration but still doesn't work.
I've searched around, seems like this is pretty common issue, but no solution yet.

Please update if you have solution for this.

**Reply by nicolas-gagnon** [reply]
I follow https://www.youtube.com/watch?v=eJSYzXRdXM0&t=118s this and i put my nic mangement static with dns 8.8.8.8 then i set ip for login for the web page also i did use virtual box and he work  But  i think i will do OVA file later one  . He was stuck at 80% maybe for 20 min then stuck at 95% for other 30 min

**Comment by TOoSmOotH** [comment, marked as answer]
2.3.21 has some improvements here. When selecting your management interface, it is important you select the one that the system uses as the default route. If you use the interface with the default route as a monitor, the TX portion will be disabled during the install making it "freeze" up.

**Reply by Nina-007** [reply]
When doing a distributed installation, with the manager node there is only one interface to select. What could be the reason it freezes up then? 

- Thanks in advance

**Reply by petiepooo** [reply]
@TOoSmOotH if the proper selection is limited to just the interface with the default route, how about we change that install step from a selection box to an information box.  "You have stated that the administrative interface is already configured, and the default route is using that interface as expected.  If you need to change this later, please edit the file at ...."

Also, I've found that one of the early installation steps clobbers the /etc/hosts file by removing the entry for the administrative interface.  I have solved the install hanging issue by manually editing the /etc/hosts file to add that entry.  There should always be an entry for the IP of the administrative interface that matches the machine name in /etc/hostname.

**Comment by jertel** [comment]
For those still stuck with installs blocked during thehive setup process, are you choosing OTHER when setup prompts you to select how you will access the web interface, and then specifying an external facing hostname/ip in the follow up prompt?

**Reply by GlennHD** [reply]
I finally got this working on Ubuntu 18.04.5. Using the "OTHER" option.  The setup script kept changing the `hostname` and `/etc/hosts` file to a non-working configuration. So whenever I hit that 80% mark, I just changed my hostname back to the FQDN (e.g., `hostnamectl set-hostname so.labnet.lan`) and added the proper entry in `/etc/hosts` (e.g., `1.2.3.4 so.labnet.lan so.labnet so` and removed all of those SO-generated entries for localhost/v6.) When I switched back to the setup script, it just started chugging past that 80%. I verified that the configuration worked (as best I could) by validating in nginx as well as making sure all integrations in the SO app properly generating the correct pivot links. So far so good....

**Reply by jakeatbgohiodotorg** [reply]
Thanks, GlennHD! Updating the HOSTS file worked perfectly.

**Reply by GuardianCosmos** [reply]
I was running into this same issue, and your fix to manually modify the hosts file and hostname worked like a charm! I was banging my head on the desk for quite a few hours trying to figure out what was going wrong haha

**Reply by innovate-support** [reply]
Using the CentOS iso this didn't work.  Nothing I do seems to work to get past the 80% Hive hanging.  The only way I ever get Hive installed is to use the v2.3.2.iso and then update after installation.  This comes with it's own issues though that I'm still working through.  After spending a full week changing options and re-installing, snap-shotting, and changing all sorts of different options and scenarios I've concluded that this v2.3.x release of SO is unstable and not ready for production.  And I'm not doing anything special either, just trying to install a Manager on one network and get a remote Sensor to send logs to it.  Distributed setups simply don't work and the docs are lacking.  These forums are very helpful but only re-enforces that there's so many problems with v2.3.x that it shouldn't be pushed as a production product. Something is broken in either the internal SO firewall or the setup of layer 4 interactions with hostnames, ports, and public vs private IP's.

**Reply by Nina-007** [reply]
Hi @GlennHD , 
How did you re-enter the setup after changing the hostname back to FQDN?

**Reply by GlennHD** [reply]
> Hi @GlennHD ,
> How did you re-enter the setup after changing the hostname back to FQDN?

I was SSH'd into the VM with two terminals. If you're using the appliance ISO, I'm not sure how to help there. I've always built on distribution releases and I typically stay away from "appliance" ISOs since I find I'm not really good at making those work 😕

**Reply by petiepooo** [reply]
@GlennHD I think he's referring to the command you would run, not how to connect to the box..

**Reply by Nina-007** [reply]
Yes @petiepooo is correct. 

I meant the command you used. 

But if you ssh'd from a different box, then I think I understand. 
I'm using the ISO

**Reply by innovate-support** [reply]
It's NOT explicitly stated in the docs, but this may help you.

To re-setup everything, even IP info:
`sudo bash /home/YOURUSER/SecurityOnion/setup/so-setup iso`

To re-setup SO stuff but leave the IP address alone.
`sudo bash /home/YOURUSER/SecurityOnion/setup/so-setup network`
or
`sudo bash /home/YOURUSER/SecurityOnion/so-setup-network`

**Reply by Nina-007** [reply]
Thanks @innovate-support. 
this helps.

**Reply by dougburks** [reply]
I've updated the Installation page:
https://docs.securityonion.net/en/2.3/installation.html
